src/test/java/io/aiven/connect/jdbc/sink/JdbcSinkConfigTest.java

```diff
@@ -89,4 +92,84 @@ public void verifyDeleteEnabled() {
+        final Map<String, String> props = new HashMap<>();
+        props.put(JdbcSinkConfig.CONNECTION_URL_CONFIG, "jdbc://localhost");
+        props.put(JdbcSinkConfig.PK_MODE, "kafka");
+        props.put(JdbcSinkConfig.PK_FIELDS, "topic,partition");
+
+        final Config config = new JdbcSinkConnector().validate(props);
+
+        assertTrue(config.configValues().stream().anyMatch(cv -> cv.errorMessages().size() > 0));
+        assertTrue(config.configValues().stream()
+                .filter(cv -> cv.name().equals(JdbcSinkConfig.PK_FIELDS))
+                .flatMap(cv -> cv.errorMessages().stream())
+                .anyMatch(msg -> msg.contains(
+                        "Primary key fields must be set with three fields "
+                                + "(topic, partition, offset) when pkMode is 'kafka'"
+                )));
+    }
+
+    @Test
+    public void shouldValidatePKModeRecordKeyWithNoPKFields() {
+        final Map<String, String> props = new HashMap<>();
+        props.put(JdbcSinkConfig.CONNECTION_URL_CONFIG, "jdbc://localhost");
+        props.put(JdbcSinkConfig.PK_MODE, "record_key");
+
+        final Config config = new JdbcSinkConnector().validate(props);
+
+        assertTrue(config.configValues().stream().anyMatch(cv -> cv.errorMessages().size() > 0));
+        assertTrue(config.configValues().stream()
+                .filter(cv -> cv.name().equals(JdbcSinkConfig.PK_FIELDS))
+                .flatMap(cv -> cv.errorMessages().stream())
+                .anyMatch(msg -> msg.contains(
+                        "Primary key fields must be set when pkMode is 'record_key' or 'record_value'"
+                )));
+    }
+
+    @Test
+    public void shouldValidatePKModeRecordValueWithNoPKFields() {
+        final Map<String, String> props = new HashMap<>();
+        props.put(JdbcSinkConfig.CONNECTION_URL_CONFIG, "jdbc://localhost");
+        props.put(JdbcSinkConfig.PK_MODE, "record_value");
+
+        final Config config = new JdbcSinkConnector().validate(props);
+
+        assertTrue(config.configValues().stream().anyMatch(cv -> cv.errorMessages().size() > 0));
+        assertTrue(config.configValues().stream()
+                .filter(cv -> cv.name().equals(JdbcSinkConfig.PK_FIELDS))
+                .flatMap(cv -> cv.errorMessages().stream())
+                .anyMatch(msg -> msg.contains(
+                        "Primary key fields must be set when pkMode is 'record_key' or 'record_value'"
+                )));
+    }
+
+    @Test
+    public void shouldValidateValidPKModeAndPKFields() {
+        final Map<String, String> props = new HashMap<>();
+        props.put(JdbcSinkConfig.CONNECTION_URL_CONFIG, "jdbc://localhost");
+        props.put(JdbcSinkConfig.PK_MODE, "record_key");
+        props.put(JdbcSinkConfig.PK_FIELDS, "id");
+
+        final Config config = new JdbcSinkConnector().validate(props);
+
+        assertFalse(config.configValues().stream().allMatch(cv -> cv.errorMessages().size() > 0));
```

Comment:
This asserts that there's at least one property that didn't have an error message. I'm guessing we actually want to assert that no properties have error messages?
```suggestion
        assertFalse(config.configValues().stream().anyMatch(cv -> cv.errorMessages().size() > 0));
```